### ISSUE TITLE
`fit_*()` function documentation

### ISSUE THREAD

**Issue by vincentarelbundock** [vincentarelbundock/pymarginaleffects#175]
@artiom-matvei  here's a minimal example

```python
from marginaleffects import *
from sklearn.linear_model import LogisticRegression

dat = get_dataset("thornton").drop_nulls()

###### standard workflow
# 1. create a y and X matrix
X = dat.select("age", "distance", "incentive").to_numpy()
y = dat["outcome"].to_numpy()
# 2. fit model
mod = LogisticRegression().fit(X, y)
# 3. Ooops, this doesn't work in `marginaleffects`


####### new workflow
mod = fit_sklearn(
    "outcome ~ age + distance + incentive", data=dat, engine=LogisticRegression
)
avg_predictions(mod)
avg_slopes(mod, variables="age")
```

**Comment by artiom-matvei** [comment]
we have 71% test coverage which is pretty good (even though plot_* tests are excluded on MacOS), I will add tests to fit_sklearn because there are currently none

![Image](https://github.com/user-attachments/assets/20f57829-ba33-431c-965e-dc8fea7eff8b)

**Comment by vincentarelbundock** [comment]
Yep, that's nice. But a lot of the tests are "dumb", without checking the actual accuracy of results against `R`. Need to think of a better strategy for this.
